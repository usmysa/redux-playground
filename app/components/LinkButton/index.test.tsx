import { renderWithProviders } from "@/__tests__";
import { fireEvent, screen, waitFor } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router";
import { LinkButton } from ".";

describe("LinkButton", () => {
  it("renders correctly", () => {
    const routes = [
      {
        path: "/",
        element: <LinkButton label="Label" to="/test" />,
      },
      {
        path: "/test",
        element: <div>TEST</div>,
      },
    ];
    const router = createMemoryRouter(routes, { initialEntries: ["/"] });
    const { container } = renderWithProviders(
      <RouterProvider router={router} />,
    );

    expect(container).toMatchSnapshot();
  });

  it("redirect to path when button is clicked", async () => {
    const routes = [
      {
        path: "/",
        element: <LinkButton label="Label" to="/test" />,
      },
      {
        path: "/test",
        element: <div>TEST</div>,
      },
    ];
    const router = createMemoryRouter(routes, { initialEntries: ["/"] });
    renderWithProviders(<RouterProvider router={router} />);

    const button = await screen.findByRole("button", { name: "Label" });
    fireEvent.click(button);

    await waitFor(() => {
      expect(router.state.location.pathname).toEqual("/test");
    });
  });
});
