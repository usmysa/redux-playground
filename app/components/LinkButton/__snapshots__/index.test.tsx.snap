// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkButton > renders correctly 1`] = `
<div>
  <a
    href="/test"
  >
    <button
      aria-label="Label"
      type="button"
    >
      Label
    </button>
  </a>
</div>
`;
